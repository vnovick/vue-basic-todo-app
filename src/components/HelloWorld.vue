<template>
  <div class="hello">
    <header>
      <slot name="header" :user="user">{{user.firstName}}</slot>
      <input type="text" v-model="html">
      <div></div>
    </header>
    <section>
      <button class="buttonSomething" @click="toggleShow">{{ caption}}</button>
      <transition
        @before-enter="beforeEnter"
        name="custom-classes-transition"
        enter-active-class="animated tada"
        leave-active-class="animated hinge"
      >
        <p v-if="show" class="hello-caption">hello</p>
      </transition>
      <slot></slot>
    </section>
    <footer>
      <input type="text" ref="otherInput"/>
      <slot name="footer">Default footer</slot>
    </footer>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  computed: {
    fullName(){
      return `${this.user.firstName} ${this.user.lastName}`
    },
    otherInput() {
      return this.$refs.otherInput
    }
  },
  data(){
    return {
      html: '',
      caption: 'Show Hello',
      myarg: 'other.prevent',
      show: false,
      user: {
        firstName: 'Vladimir',
        lastName: 'Novick'
      }
    }
  },
  methods: {
    beforeEnter(){
      console.log("Gonna animate")
    },
    toggleShow(){
      this.caption = 'Hide Hello'
      this.show = !this.show
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
	
.bounce-enter-active {
  animation: bounce-in .5s;
}
.bounce-leave-active {
  animation: bounce-in .5s reverse;
}
@keyframes bounce-in {
  0% {
    transform: scale(0);
  }
  50% {
    transform: scale(1.5);
  }
  100% {
    transform: scale(1);
  }
}
</style>
