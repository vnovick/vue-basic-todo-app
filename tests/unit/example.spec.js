import { mount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe('Hello World component test', () => {

    //Snapshot test
    it('Check that markup is correct', () => {
    const msg = 'new message'
    const wrapper = mount(HelloWorld, {
      propsData: { msg },
    })
    expect(wrapper).toMatchSnapshot()
  })
  // Unit test
  it('Check that it has button that triggers toggleShow', () => {
    const msg = 'new message'
    const toggleShow = jest.fn()
    const wrapper = mount(HelloWorld, {
      propsData: { msg },
      methods: {
        toggleShow
      }
    })
    const coolButton = wrapper.find('.buttonSomething');
    coolButton.trigger('click')
    expect(toggleShow).toBeCalled()
  })

  // Snapshot test with some functionality
  it('Check that toggleShow button renders correct snapshot', () => {
    const msg = 'new message'
    const wrapper = mount(HelloWorld, {
      propsData: { msg },
    })
    const coolButton = wrapper.find('.buttonSomething');
    coolButton.trigger('click')
    const animatedHelloElement = wrapper.find('.hello-caption')
    expect(animatedHelloElement).toBeDefined();
    expect(animatedHelloElement.text()).toMatch('hello')
    expect(wrapper).toMatchSnapshot()
  })

  it('Check if slots are applied', () => {
    const wrapper = mount(HelloWorld, {
      slots: {
        default: `<div class="default-test-slot"></div>`,
        header: `<div class="header-test-slot"></div>`,
        footer: `<div class="footer-test-slot"></div>`
      }
    })

    expect(wrapper.find('.footer-test-slot')).toBeDefined()
    expect(wrapper.find('.header-test-slot')).toBeDefined()
    expect(wrapper.find('.default-test-slot')).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
})
