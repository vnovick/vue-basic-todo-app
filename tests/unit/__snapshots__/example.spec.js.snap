// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World component test Check if slots are applied 1`] = `
<div class="hello">
  <header>
    <div class="header-test-slot"></div> <input type="text">
    <div></div>
  </header>
  <section><button class="buttonSomething">Show Hello</button>
    <!---->
    <div class="default-test-slot"></div>
  </section>
  <footer><input type="text">
    <div class="footer-test-slot"></div>
  </footer>
</div>
`;

exports[`Hello World component test Check that markup is correct 1`] = `
<div class="hello" msg="new message">
  <header>Vladimir <input type="text">
    <div></div>
  </header>
  <section><button class="buttonSomething">Show Hello</button>
    <!---->
  </section>
  <footer><input type="text"> Default footer</footer>
</div>
`;

exports[`Hello World component test Check that toggleShow button renders correct snapshot 1`] = `
<div class="hello" msg="new message">
  <header>Vladimir <input type="text">
    <div></div>
  </header>
  <section><button class="buttonSomething">Hide Hello</button>
    <p class="hello-caption" name="custom-classes-transition" enter-active-class="animated tada" leave-active-class="animated hinge">hello</p>
  </section>
  <footer><input type="text"> Default footer</footer>
</div>
`;
